### PR TITLE
BS-4066: Requested changes.

### PR DESCRIPTION
**Pull request by yashtewari**
These changes were requested in #19 but postponed due to priorities.

### CODE REVIEW

File: src/main/java/security/build/pdp/client/PDPInterceptor.java [hunk 1]
```diff
@@ -99,19 +129,29 @@ public boolean getInterceptAllEndpoints() {
         return this.ignoreRegex;
     }
 
-    public void setEnable(boolean enable) {
+    public void setEnable(boolean enable) throws IOException {
         this.enable = enable;
+
+        checkConfigurationIntegrity();
```
Comment (eyalkraft) [inline src/main/java/security/build/pdp/client/PDPInterceptor.java:135]: this should happen before setting the enable flag

File: src/main/java/security/build/pdp/client/PDPInterceptor.java [hunk 2]
```diff
@@ -99,19 +129,29 @@ public boolean getInterceptAllEndpoints() {
...
     }
 
-    public void setInterceptAllEndpoints(boolean interceptAllEndpoints) {
+    public void setInterceptAllEndpoints(boolean interceptAllEndpoints) throws IOException {
```
Comment (eyalkraft) [inline src/main/java/security/build/pdp/client/PDPInterceptor.java:138]: generally all setters currently succeed even if they set illegal values.

**Review by eyalkraft — CHANGES_REQUESTED** [review]
👏

File: src/main/java/security/build/pdp/client/PDPInterceptor.java (hunk 2 above)
Comment (yashtewari) [reply src/main/java/security/build/pdp/client/PDPInterceptor.java:144]: Fixed

**Review by eyalkraft — APPROVED** [review]

**Pull request merged**